drivers/hud_twenty_twenty_two_to_twenty_twenty_four/app/models/hud_twenty_twenty_two_to_twenty_twenty_four/csv_transformer.rb

```diff
@@ -0,0 +1,230 @@
+      'AssessmentQuestions.csv' => {
+        action: :copy,
+        model: GrdaWarehouse::Hud::AssessmentQuestion,
+      },
+      'AssessmentResults.csv' => {
+        action: :copy,
+        model: GrdaWarehouse::Hud::AssessmentResult,
+      },
+      'Event.csv' => {
+        action: :copy,
+        model: GrdaWarehouse::Hud::Event,
+      },
+      'YouthEducationStatus.csv' => {
+        action: :copy,
+        model: GrdaWarehouse::Hud::YouthEducationStatus,
+      },
+    }.freeze
+
+    def self.up(source_directory, destination_directory)
+      TRANSFORM_TYPES.each do |file, transform|
+        action = transform[:action]
+        model = transform[:model]
+        transformer = transform[:transformer]
+        references = transform[:references] || {}
+
+        source_file = File.join(source_directory, file)
+        destination_file = File.join(destination_directory, file)
+        next unless action == :create || File.exist?(source_file)
+
+        references.transform_values! do |reference|
+          file = File.join(source_directory, reference[:file])
+          next unless File.exist?(file)
+
+          encoding = AutoEncodingCsv.detect_encoding(file)
+          fix_bad_line_endings(file, encoding)
+          {
+            file: file,
+            model: reference[:model],
+          }
+        end
+
+        case action
+        when :copy
+          encoding = AutoEncodingCsv.detect_encoding(source_file)
+          fix_bad_line_endings(source_file, encoding)
+          FileUtils.copy(source_file, destination_file)
+        when :update
+          encoding = AutoEncodingCsv.detect_encoding(source_file)
+          fix_bad_line_endings(source_file, encoding)
+          ::Kiba.run(transformer.up(source_file, destination_file, encoding, header_converter(model), references))
+        when :create
+          if transformer.present?
+            ::Kiba.run(transformer.create(destination_file, references))
+          else
+            CSV.open(
+              destination_file, 'w',
+              write_headers: true,
+              headers: model.hmis_configuration(version: '2024').keys.map(&:to_s)
+            ) { |csv| } # Empty block
+          end
+        end
+      end
+    end
+
+    def self.fix_bad_line_endings(filename, encoding)
+      tmp_file = ::Tempfile.new(filename)
+      file_with_bad_line_endings = false
+
+      File.open(filename, 'r', encoding: encoding) do |file|
+        file_with_bad_line_endings = ! valid_line_endings?(file)
+      end
+
+      if file_with_bad_line_endings
+        File.open(filename, 'r', encoding: encoding) do |file|
+          copy_length = file.stat.size - 2
+          Rails.logger.debug "Correcting bad line ending in #{filename}"
+          File.copy_stream(file, tmp_file, copy_length, 0)
+          tmp_file.write("\n")
+          tmp_file.close
+        end
+        FileUtils.cp(tmp_file, filename)
+      end
+    ensure
+      tmp_file&.close
+      tmp_file&.unlink
+    end
```

Comment:
What do you think about moving this logic up in to `/app/models/something` or maybe as a monkey patch for CSV?  I feel like we call this pretty often, might be nice to centralize it.